src/legacy/core_plugins/data/public/actions/select_range_action.ts

```diff
@@ -68,21 +59,23 @@ export function selectRangeAction(
         throw new IncompatibleActionError();
       }
 
-      const filters: Filter[] = (await onBrushEvent(data, getIndexPatterns)) || [];
+      const filter = await onBrushEvent(data);
 
-      const selectedFilters: Filter[] = esFilters.mapAndFlattenFilters(filters);
+      if (filter) {
```

Comment:
nit: I like the cleanup here, but [based on our styleguide](https://github.com/elastic/kibana/blob/master/STYLEGUIDE.md#returnthrow-early-from-functions) the preferred approach would be to return early, e.g.

```js
if (!filter) {
  return;
}

const selectedFilters: .....
```

Reply:
changed.